crates/runtime/src/databackend/memtable.rs

```diff
@@ -24,7 +24,7 @@ impl MemTableBackend {
     pub fn new(ctx: Arc<SessionContext>, name: &str) -> Self {
         MemTableBackend {
             ctx,
-            name: name.replace('.', "_").to_string(),
+            name: crate::datafusion::get_internal_dataset_name(name),
```

Comment:
With the new model of `from` - I don't think we need to do this anymore

Reply:
We probably need some validation on valid names though (and to spec what is valid). Probably whatever is a valid GitHub repo and org name is a good start.

Reply:
Deferring discussion of this to the discussion on the still open PR: https://github.com/spiceai/spiceai/pull/632#discussion_r1472242352